cloudprovider/aws/aws_provider.go

```diff
@@ -0,0 +1,70 @@
+package aws
+
+import (
+	"encoding/json"
+	"github.com/aws/aws-sdk-go/aws/ec2metadata"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/rancher/agent/utilities/config"
+	"os"
+	"path"
+	"time"
+)
+
+const (
+	regionLabel           = "ec2.region"
+	availabilityZoneLabel = "ec2.availability_zone"
+	location              = "aws"
+	infoFile              = "info.json"
+	tempFile              = "temp.json"
+)
+
+type Info struct {
+	Label map[string]string `json:"label,omitempty"`
+}
+
+func CallProvider() {
+	infoPath := path.Join(config.StateDir(), location, infoFile)
+	tempPath := path.Join(config.StateDir(), location, tempFile)
+	select {
+	case <-time.After(time.Minute * 5):
+		return
+	default:
+		for {
+			if _, err := os.Stat(infoPath); err == nil {
+				return
+			}
+			time.Sleep(time.Second * 10)
+			session, err := session.NewSession()
```

Comment:
Do you have to create a new session and md client every loop or could you create them once outside the loop and reuse?